daft/dataframe/dataframe.py

```diff
@@ -1347,6 +1363,30 @@ def select(self, *columns: ColumnInputType) -> "DataFrame":
         builder = self._builder.select(self.__column_input_to_expression(columns))
         return DataFrame(builder)
 
+    @DataframePublicAPI
+    def describe(self) -> "DataFrame":
+        """Returns the Schema of the DataFrame, which provides information about each column, as a new DataFrame.
+
+        Example:
+        >>> import daft
+        >>> df = daft.from_pydict({"a": [1, 2, 3], "b": ["x", "y", "z"]})
+        >>> df.describe().show()
+            ╭────────┬───────╮
+            │ column ┆ type  │
+            │ ---    ┆ ---   │
```

Comment:
For reviewers, thoughts on "name" vs "column" as the column name for the column's name.

- ANSI-SQL specifies "name"
- postgres uses "column"
- duckdb uses "column_name"
- hive uses "col_name"
- spark uses "col_name"

I chose "column" because it's more descriptive than "name" (IMO) and consistent with PostgreSQL.

Reply:
I personally like "column_name" as I think it's the most descriptive

Reply:
FWIW, datafusion also uses `column_name`

```sql
DataFusion CLI v40.0.0
> create table t1 as select 1;
0 row(s) fetched.
Elapsed 0.036 seconds.

> describe t1;
+-------------+-----------+-------------+
| column_name | data_type | is_nullable |
+-------------+-----------+-------------+
| Int64(1)    | Int64     | NO          |
+-------------+-----------+-------------+
1 row(s) fetched.
Elapsed 0.006 seconds.

```